src/components/ae-identicon/ae-identicon.vue

```diff
@@ -0,0 +1,83 @@
+<template>
+  <canvas
+    :class="['ae-identicon', { shadow }]"
+    ref="blockie"
+  />
+</template>
+
+<script>
+import { renderIcon } from '@download/blockies';
+
+/**
+ * Displays the representation of an aeternity address as an identicon
+ * @see https://npmjs.com/package/@download/blockies
+ */
+export default {
+  name: 'ae-identicon',
+  props: {
+    /**
+     * Address of the wallet for which the identicon is going to generate a blockies from.
+     */
+    address: {
+      type: String,
+      required: true,
+    },
+    /**
+     * Set the identicon size, valid sizes: `xl, base, s, xs, xxs`.
+     * The default value us: `base`
+     */
+    size: {
+      type: String,
+      validator: value => [
+        'xl',
+        'base',
+        's',
+        'xs',
+        'xxs',
+      ].includes(value),
+      default: 'base',
+    },
+
+    /**
+     * Apply shadow to the identicon
+     */
+    shadow: Boolean,
+  },
+  watch: {
+    address() {
+      this.render();
+    },
+  },
+  mounted() {
+    this.render();
+  },
+  methods: {
+    render() {
+      renderIcon({
+        seed: this.address,
+        size: 16,
+        scale: ({
+          xl: 6,
+          base: 2,
+          s: 1.5,
+          xs: 1.25,
+          xxs: 0.875,
+        })[this.size],
```

Comment:
@davidyuk I know, but in this case the division works, because 1rem = 16px, thus we set the initial size of the component to 16px and then increment or decrement based on the scale.

> Width and height should be set in styles by rem units instead of using scale param.

I don't think the canvas works like that. You can set the canvas width / height but the rendered elements inside will be the same size you've generated the 2d canvas, and would crop out the avatar.

A solution for this would be:

- Set the width / height dynamically on avatar generation with the canvas
- Have the avatar be an image output (.png, jpg) and set the width / height dynamically with styles.

Reply:
Please, check this example:
`<html styles="font-size: 16px">`:
![screenshot from 2018-10-09 19-31-00](https://user-images.githubusercontent.com/9007851/46686924-df647c80-cbf9-11e8-9e4b-49400617599b.png)
`<html styles="font-size: 30px">`:
![screenshot from 2018-10-09 19-34-22](https://user-images.githubusercontent.com/9007851/46687115-58fc6a80-cbfa-11e8-8354-124e677f578c.png)

> You can set the canvas width / height but the rendered elements inside will be the same size you've generated the 2d canvas, and would crop out the avatar

no, it stretches

Reply:
should be better now.